### DISCUSSION TITLE
Hobby account cookie limitation

### DISCUSSION THREAD

**Discussion by andrevenancio** [vercel/community#4925] [Help] [answered]
### Link to the code that reproduces this issue

https://github.com/andrevenancio/nextjs-cookies

### Vercel app

https://nextjs-cookies.vercel.app/

### To Reproduce

This repository works well for `Hobby`, `Pro` and `Enterprise` accounts locally.

Hoever when I deploy to vercel, there are unexpected results depending on your account type.

Lets imagine the the deployment url from vercel is test123.vercel.app, and you are always accessing this URL from a incognito tab on your google chrome. Meaning you are NOT logged in to vercel.com using any of your accounts, you are simply accessing the application URL.

If the `test123.vercel.app` belongs to a hobby account, the url redirects to a vercel.com domain and tries to connect the user. If the `test123.vercel.app` belongs to a Enterprise account, everything works as expected.

I'm assuming that there is some undocumented check's on Hobby accounts to make sure users pay for `Advanced Deployment protection` costing 150$ a month. Which if true, is fine, its your cloud infrastructure, your business model, but that information should probably be explicit on the documentation.

### Current vs. Expected behavior

I'm expecting the same code works across any account type.

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.1.0: Mon Oct  9 21:28:45 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6020
Binaries:
  Node: 18.18.2
  npm: 9.8.1
  Yarn: 1.22.19
  pnpm: 8.9.2
Relevant Packages:
  next: 14.0.4-canary.5
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Additional context

I would like to understand if this is a check that happens on hobby accounts in order to make sure the advanced deployment protection is the only option users on hobby accounts have to protect their domains with a password. Again, if this is the business model fine, but would be great to be this written somewhere on the documentation

**Comment by leerob** [comment, marked as answer]
There is not any special logic here based on plans. We also have an example for a Middleware version of authentication: https://vercel.com/templates/next.js/basic-auth-password

**Comment by andrevenancio** [comment]
Hi @leerob, thanks for your reply. I think I was missing some context in my description.
If I deploy the application above in my hobby account and I go with the vercel generated url (https://nextjs-cookies.vercel.app/) then the application works as expected.

But I'm going to do a couple of things:
1) I've created a new branch on github called `develop` which is now the default branch on my github project (https://github.com/andrevenancio/nextjs-cookies)
2) On vercel's dashboard I had a env variable named `NEXT_PUBLIC_PASSWORD_PROTECTED` with the value `letmein`. That env variable is now only on `Preview on develop`. 
3) I push a new code change to `develop`, that automatically creates a new preview url to be generated. this is the [preview url](https://nextjs-cookies-kojk2y2mr-andrevenancio.vercel.app/)

Now if you try to open this preview route on a incognito window there is a redirect in place that forces me to login to vercel.com.
I'm I access this URL from a chrome profile that is logged in to vercel.com, then I can see the page.

I assumed that this might be because I still have "comments" active for this project (it makes sense to have to be connected to comment on the preview route) but even with them Off I am still forced to login to vercel.

I assume that any preview route that is NOT done against the production branch `main` will always behave like this?

However, I want to add a custom domain.
www.mywebsite.com is my production URL linked to my `main` branch on github
develop.mywebsite.com is my development URL linked to my `develop` branch on github.

I dont want users testing my dev site to have to be connected to vercel. maybe we can argue that for a development/production case like the one I just illustrated that makes sense, but I might just want to create a subdomain and make it accessible to the public.

lets say:

www.news.com
portugal.news.com

and if I use preview branches then accessing portugal.news.com will only happen if I'm logged in on vercel.

So maybe this is all a big misunderstanding on how to use multiple subdomains on the same project, would love to get more insight into how to resolve this properly.
Thank you

**Reply by leerob** [reply]
Here's how we recommend using subdomains 😄 

https://vercel.com/guides/nextjs-multi-tenant-application
